Source/WebCore/rendering/HitTestResult.cpp

```diff
@@ -809,19 +795,19 @@ Vector<String> HitTestResult::dictationAlternatives() const
 Element* HitTestResult::targetElement() const
 {
     for (Node* node = m_innerNode.get(); node; node = node->parentInComposedTree()) {
-        if (is<Element>(*node))
-            return downcast<Element>(node);
+        if (auto* element = dynamicDowncast<Element>(*node))
+            return element;
     }
     return nullptr;
 }
 
 Element* HitTestResult::innerNonSharedElement() const
 {
-    Node* node = m_innerNonSharedNode.get();
+    auto* node = m_innerNonSharedNode.get();
     if (!node)
         return nullptr;
-    if (is<Element>(*node))
-        return downcast<Element>(node);
+    if (auto* element = dynamicDowncast<Element>(*node))
+        return element;
     return node->parentElement();
```

Comment:
Seems like this operation "self if element, parent element otherwise" should be a named function.